src/registrar/templates/domain_request_review.html

```diff
@@ -155,11 +155,8 @@ <h3 class="register-form-review-header">Alternative domains</h3>
         {% endif %}
 
 
-        {% if step == Step.ANYTHING_ELSE %}
-          {% namespaced_url 'domain-request' step as domain_request_url %}
-          {% with title=form_titles|get_item:step value=domain_request.anything_else|default:"No" %}
-            {% include "includes/summary_item.html" with title=title value=value heading_level=heading_level editable=True edit_link=domain_request_url %}
-          {% endwith %}
+        {% if step == Step.ADDITIONAL_DETAILS %}
```

Comment:
See comment below. I think you simplified here at the cost of a new repetitive block.